solutions/problem4.js

```diff
@@ -0,0 +1,17 @@
+const { isArrayEmpty, isArrayFull } = require('./utils')
+
+const takeFirst = (count, ...promises) => isArrayEmpty(promises)
+  ? Promise.resolve([])
+  : new Promise((resolve, reject) => {
+    const results = []
+
+    return promises.map(promise => promise
+      .then(val => results.push(val) 
+        && isArrayFull(results, count) 
```

Comment:
This first `&&` is unnecessary imho. Just makes code more complicated than it should be. First statement will be always truthy anyways.